src/common/commonutils/FileUtils.c

```diff
@@ -120,25 +120,52 @@ bool SavePayloadToFile(const char* fileName, const char* payload, const int payl
 
 bool AppendPayloadToFile(const char* fileName, const char* payload, const int payloadSizeBytes, void* log)
 {
-    char* fileContents = NULL;
+    FILE* existingFile = NULL;
+    struct stat st;
     bool result = false;
+    char lastCharacter = EOF;
+
+    if(NULL == fileName || NULL == payload || 0 == payloadSizeBytes)
+    {
+        OsConfigLogError(log, "AppendPayloadToFile: invalid arguments");
+        return false;
+    }
 
-    // If the file exists and there is no EOL at the end of file, add one before the append
-    if ((NULL != payload) && (payloadSizeBytes > 0) && FileExists(fileName) && 
-        (NULL != (fileContents = LoadStringFromFile(fileName, false, log))) && 
-        (EOL != fileContents[strlen(fileContents) - 1]))
+    /* If the file exists and there is no EOL at the end of file, add one before the append. */
+    if (stat(fileName, &st) == 0 && st.st_size > 0)

```

Comment:
>stat(fileName, &st) == 0 && st.st_size > 0) [](http://example.com/codeflow?start=8&length=43)

Inner parenthesis and move constant to the left in comparisons to keep with existing code style